From 405ab6c949297645fe2a7c28281e9f855c632e75 Mon Sep 17 00:00:00 2001
From: Ramneek Mehresh <ramneek.mehresh@freescale.com>
Date: Tue, 28 May 2013 13:52:32 +0530
Subject: [PATCH 073/124] fsl/otg: Resolve OTG crash issue with another host

Resolves kernel crash issue when a USB flash drive is inserted
into USB1 port with USB2 port configured as otg

Signed-off-by: Ramneek Mehresh <ramneek.mehresh@freescale.com>
Change-Id: I6cd53d4fd47c1a77197da6f41dcbd0bd63b7198d
Reviewed-on: http://git.am.freescale.net:8181/2722
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Gupta Suresh-B42813 <suresh.gupta@freescale.com>
Reviewed-by: Aggrwal Poonam-B10812 <Poonam.Aggrwal@freescale.com>
Reviewed-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
---
 drivers/usb/core/hub.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/usb/core/hub.c b/drivers/usb/core/hub.c
index 563d84e..8385f22 100644
--- a/drivers/usb/core/hub.c
+++ b/drivers/usb/core/hub.c
@@ -4306,11 +4306,13 @@ hub_port_init (struct usb_hub *hub, struct usb_device *udev, int port1,
 	else
 		speed = usb_speed_string(udev->speed);
 
+#if !defined(CONFIG_FSL_USB2_OTG) && !defined(CONFIG_FSL_USB2_OTG_MODULE)
 	if (udev->speed != USB_SPEED_SUPER)
 		dev_info(&udev->dev,
 				"%s %s USB device number %d using %s\n",
 				(udev->config) ? "reset" : "new", speed,
 				devnum, udev->bus->controller->driver->name);
+#endif
 
 	/* Set up TT records, if needed  */
 	if (hdev->tt) {
-- 
2.1.0.27.g96db324

